### PR TITLE
some clippy fixes

### PR DESCRIPTION
**Pull request by matthiaskrgr**
I used rustfmt-nightly to format the changes in the latest commit, is that correct or do I have to use git-master rustfmt?

### CODE REVIEW

**Comment by calebcartwright** [comment]
> is that correct or do I have to use git-master rustfmt?

The self-tests run against the codebase with the same rustfmt produced by that codebase, so using a different version of rustfmt (like 1.4.20 on nightly) to format the code on the master branch will indeed be problematic.

You may want to revert bd1203db07da768f468d340e944de991992b690c and then just tweak your changes as necessary to ensure the self tests pass to validate formatting

**Comment by matthiaskrgr** [comment]
```
CFG_RELEASE_CHANNEL="" CFG_RELEASE=""  cargo build --features rustfmt,cargo-fmt
./target/debug/cargo-fmt --all
```
throws an error: `Unrecognized option: 'recursive'`

**Comment by calebcartwright** [comment]
You need to run the rustfmt bin, not cargo fmt

**Comment by matthiaskrgr** [comment]
It's not working, I only get
```
$./target/debug/rustfmt  src/formatting/macros.rs
Warning: can't set `error_on_line_overflow = true`, unstable features can only be used on stable or beta when `unstable_features` is also enabled.
Warning: can't set `error_on_unformatted = true`, unstable features can only be used on stable or beta when `unstable_features` is also enabled.
```

And the file is unchanged

**Comment by calebcartwright** [comment]
You need to use `cargo run`. Sorry, I was on my phone earlier and didn't realize you weren't doing this before

For example (drop the `--check` flag if you need to fix some formatting):
```bash
cargo +nightly run --bin rustfmt --features rustfmt -- src/lib.rs --check --recursive
```

File: config_proc_macro/src/attrs.rs [hunk 1]
```diff
@@ -39,10 +39,7 @@ pub fn is_config_value(attr: &syn::Attribute) -> bool {
 }
 
 fn is_attr_name_value(attr: &syn::Attribute, name: &str) -> bool {
-    attr.parse_meta().ok().map_or(false, |meta| match meta {
-        syn::Meta::NameValue(syn::MetaNameValue { ref path, .. }) if path.is_ident(name) => true,
-        _ => false,
-    })
+    attr.parse_meta().ok().map_or(false, |meta| matches!(meta, syn::Meta::NameValue(syn::MetaNameValue { ref path, .. }) if path.is_ident(name)))
```
Comment (calebcartwright) [inline config_proc_macro/src/attrs.rs:42]: This is not being formatted due to the chain length issue and would need to be hand formatted. We can address this later though

File: src/formatting/closures.rs [hunk 2]
```diff
@@ -124,19 +124,19 @@ fn needs_block(block: &ast::Block, prefix: &str, context: &RewriteContext<'_>) -
...
+            | ast::ExprKind::Binary(..)
+            | ast::ExprKind::Cast(..)
+            | ast::ExprKind::Type(..)
+            | ast::ExprKind::Assign(..)
+            | ast::ExprKind::AssignOp(..)
+            | ast::ExprKind::Field(..)
+            | ast::ExprKind::Index(..)
+            | ast::ExprKind::Range(..)
+            | ast::ExprKind::Try(..)
+    )
```
Comment (calebcartwright) [inline src/formatting/closures.rs:139]: I take it this is coming from `match_like_matches_macro`? TBH, I find the original match usage to be more readable. Using the macro and the resulting the extra indentation just looks odd

**Comment by calebcartwright** [comment]
Personally I feel like a few of these make the code trickier to read, particular the breaking of visual alignment in cases that had higher quantities of match arm patterns, but if that's what clippy wants then I'm okay with it :smile:

File: src/formatting/closures.rs (hunk 2 above)
Comment (calebcartwright) [reply src/formatting/closures.rs:139]: D'oh sorry, see you have the lint names in the commit message

**Pull request merged**